src/library-aux/scala/Any.scala

```diff
@@ -138,13 +138,87 @@ abstract class Any {
+   *  of the receiver object to `T0` is unspecified and should be considered an implementation detail of the 
+   *  corresponding compiler backend and can vary between target platform and the target version.
+   *  Transitioning to T0 might involve inserting conversions (including boxing/unboxing), casts, default values or
+   *  just no real operation at all.
+   *
+   *  Following some legitimate usages of `asInstanceOf`:
+   *
+   *  - turn a type into an opaque type (idiomatically done in the apply function of the companion
+   *    object of the opaque type)
+   *
+   *  - tell the compiler to treat the receiver object as `T0` after manual type checking (e.g. if flow analysis 
+   *    fails to deduce `T0` automatically after manual type checking). 
+   *
+   *  - tell the compiler that a higher-kinded type with a wildcard argument is of a certain type and due to performance
+   *    reasons you don't want to type test every element (type testing if the receiver object is e.g. a `List[Int]` is
+   *    not possible due to type erasure). Be confident that your type reasoning is correct, otherwise you might end up 
+   *    with an undefined outcome which could blow up at other places unexpectedly.
+   *
+   *  - tell the compiler that the receiver object has the same type as a type parameter or abstract type member in 
+   *    case the compiler is not able to know or infer the relation correctly.
+   *    You should be confident about your type reason. Also in this particular case type testing cannot help you due to
+   *    type erasure. But again, you should be very confident about your type reasoning.
+   *
+   *  Following some examples where the usage of asInstanceOf is discouraged:
+   *
+   *  - if `T0` is a primitive type
+   *    -> use pattern matching or [isInstanceOf] for type testing because `asInstanceOf` might insert a conversion
+   *       instead of a type check
+   *    -> use the corresponding x.toT functions (x.toChar, x.toByte etc.) if you want to convert
+   *
+   *  - load a class (e.g. Class.forName, ctx.getBean("bean-name") etc.) dynamically and then use asInstanceOf to tell 
+   *    the compiler what it should be.
+   *    -> use pattern matching or [isInstanceOf] for type testing so that you do not depend on implementation details
+   *         
+   *  In general, this method is safe if the subtype relation between the receiver object and `T0` holds
+   *  but should be considered unsafe in terms of type safety otherwise.
+   *
+   *  Following two examples where `asInstanceOf` was not used as intended and the result is most likely not as expected
+   *  {{{
+   *  val anInt = fromThirdPartyLibrary()
+   *  val d = anInt.asInstanceOf[Double]
+   *  //            ^^^ 
+   *  //            skips type safety, implementation detail: converts to double if 
+   *  //            anInt is a primitive Int, throws a ClastCastException otherwise 
```

Comment:
As I mentioned before: the *conversion* when `anInt` is statically an `Int` is *not* an implementation detail. It is as specified. (Yes, that specification is super weird, but it's there anyway.) The CCE is indeed an implementation detail, and is Scala.js fullLink is an existing example that does not result in a CCE.

Reply:
I know, I figured I am not going into platform details and it is better if the user considers it as an implementation detail. Otherwise it get's complicated again. Do you think the following would be better?
```suggestion
   *  //            skips type safety, converts to double if 
   *  //            anInt is a primitive Int, JDK implementation detail: throws a ClastCastException otherwise 
```

Reply:
That's better, yes.